### DISCUSSION TITLE
Welcome to Sonic Worlds Next Discussions!

### DISCUSSION THREAD

**Discussion by Techokami** [Techokami/SonicWorldsNext#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to provide direct feedback and discussion about this project outside of SFGHQ and Sonic Retro. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  If you have encountered a bug, please consider using the Issues page instead! Discussions is more for using the engine, not bug reporting.

**Comment by DimensionWarped** [comment]
I'd strongly recommend adding a supported versions of Godot entry to the readme.md file, mostly on account of having seen projects that work in Godot 3 and fail in Godot 4. I'm less certain about whether or not minor versions can make a significant difference.

**Reply by Techokami** [reply]
We've initially been working with Godot 4, but backported to Godot 3 for supporting the current stable release. I'll be updating the readme in a moment to address this. Thanks for pointing this out!

**Comment by DimensionWarped** [comment]
I'm not sure how aggressively to report bugs right now since I'm guessing a lot of stuff is known and the project is new. Specifically, right now I'm a little focused on a weird and awkward loss of control when sliding backwards to due losing velocity when ascending a slope. Do you want issues for pretty much everything or just really bad stuff like crashes/exceptions?

**Reply by Renhoex** [reply]
That's a quirk of the originals so that the player doesn't get stuck on curves, I would go for the sonic 3 approach where it just adds more velocity to you in the other direction but I don't have the exact variables.

Right now we're just looking for real issues with the game that would improve the gameplay / accuracy to the originals if it was fixed.

**Reply by DimensionWarped** [reply]
Well, to elaborate, put yourself right here and try running back and forth between these two hills for a bit (no rolling) and I think you'll get the feel for what I mean. Sliding while running up hill leaves you stuck facing in the direction you were initially running up the hill in after the slide... looks really weird, feels like crap. And there's this huge delay when you get 'turned around' in velocity where you can't control anything. I checked the acceleration against Sonic 2 and while I did notice that Sonic 2 does in fact not immediately let you take control, the acceleration that it pushes you down the hill in after losing velocity is much faster.

![image](https://user-images.githubusercontent.com/7477617/208857057-9de43e67-63f4-41bc-9809-8c1c649ba708.png)

**Comment by Renhoex** [comment]
I see, I'll take a look see what we can do, might be a while before it's fixed since this area of the code's tricky and I gotta find a better implementation for it

**Reply by DimensionWarped** [reply]
The first thing I would recommend is if the player is in this loss of control state and their direction of control goes against their facing direction, immediately restore full control. I don't know the code well enough to do this myself just yet, but if I figure it out I'll submit a pull request.

**Reply by DimensionWarped** [reply]
[ideal_back_and_forth_spot.zip](https://github.com/Techokami/SonicWorldsNext/files/10498453/ideal_back_and_forth_spot.zip)

Here's a save state (made in Kega Fusion, not sure if that's compatible with anything else) for Sonic 3 and Knuckles with an ideal place to compare the motion of the spot I referenced before and debug mode turned on.

EDIT: If not, location is Hydrocity Zone act 2, position is 1782, 01d2
